cert/spec.go

```diff
@@ -518,23 +509,57 @@ func (spec *Spec) renewPKI(ca *x509.Certificate) error {
 			time.Sleep(backoff)
 			continue
 		}
+		keyData, err := encodeKeyToPem(pair.PrivateKey)
+		if err != nil {
+			return err
+		}
 
 		spec.ResetBackoff()
-		spec.expiry.Cert = spec.Certificate().NotAfter
+		err = spec.Cert.File.WriteFile(
+			pem.EncodeToMemory(
```

Comment:
Again (for later PR) I think it would be nice to factor out some of the raw PEM manipulation logic into other functions to reduce complexity.

Reply:
I'll follow that up shortly in another PR.